src/react_barchart.jsx

```diff
@@ -0,0 +1,237 @@
+import ReactDOM from 'react-dom'
+import React, { useState } from 'react';
+import Suspense from './components/Suspense'
+import { ChartJS } from 'chart.js/auto'
+import { Bar } from 'react-chartjs-2'
+
+
+export default function ReactBarGraph(props) {
+
+  const [orientation, setOrientation] = useState("horizontal")
+
+  function swapOrientation() {
+    if (orientation === "horizontal") {
+      setOrientation("vertical")
+    } else {
+      setOrientation("horizontal")
+    }
+  }
+
+  const data = props.data
+
+  const ids = new Array(data.length)
+  const scores = new Array(data.length)
+  const colors = new Array(data.length)
+
+  // deconstructing the JSON object into ids and scores
+  for (var i = 0; i < data.length; i++) {
+    ids[i] = data[i].id
+    scores[i] = data[i].score
+  }
+
+  var palette = 
```

Comment:
`const` and move this outside, might also be nice to allow custom palette via props